### PR TITLE
Update documentation with intersphinx

### PR DESCRIPTION
**Pull request by sephib**
Update documentation with intersphinx project linking capabilities for Pandas and Numpy.
as suggested in issue #2589

### CODE REVIEW

**Comment by mrocklin** [comment]
Whoa, nice.  I'm looking forward to trying this out.

**Comment by mrocklin** [comment]
I suspect that @TomAugspurger has experience with intersphinx.  He might be good to review this.

In general this worked nicely on my machine.  This might encourage us to link back to the relevant numpy/pandas functions more frequently in docstrings.

File: dask/array/core.py [hunk 1]
```diff
@@ -3049,7 +3049,7 @@ def broadcast_to(x, shape, chunks=None):
 
     See Also
     --------
-    numpy.broadcast_to
+    :function:`numpy.broadcast_to`
```
Comment (TomAugspurger) [inline dask/array/core.py]: Do you know if `:function:` is the same as `:func`?

Comment (TomAugspurger) [reply dask/array/core.py]: Also, is including these the `:function:` role necessary to get the linkn? I thought that numpydoc handles these, though I may be wrong.

File: dask/array/percentile.py [hunk 2]
```diff
@@ -68,15 +68,15 @@ def merge_percentiles(finalq, qs, vals, Ns, interpolation='lower'):
 
     finalq : numpy.array
         Percentiles to compute (must use same scale as ``qs``).
-    qs : sequence of numpy.arrays
+    qs : sequence of :function:`numpy.array`s
```
Comment (TomAugspurger) [inline dask/array/percentile.py]: Should probably be `:class:`numpy.ndarray`, or just say "sequence of numpy arrays.

File: dask/array/percentile.py [hunk 3]
```diff
@@ -68,15 +68,15 @@ def merge_percentiles(finalq, qs, vals, Ns, interpolation='lower'):
...
         Percentiles calculated on different sets of data.
-    vals : sequence of numpy.arrays
+    vals : sequence of :function:`numpy.array`s
```
Comment (TomAugspurger) [inline dask/array/percentile.py]: Same comment about ndarrays.

File: dask/base.py [hunk 4]
```diff
@@ -112,7 +112,7 @@ def compute(self, **kwargs):
         """Compute this dask collection
 
         This turns a lazy Dask collection into its in-memory equivalent.
-        For example a Dask.array turns into a NumPy array and a Dask.dataframe
+        For example a Dask.array turns into a  :function:`numpy.array` and a Dask.dataframe
```
Comment (TomAugspurger) [inline dask/base.py]: Same comment about arrays.

File: dask/base.py [hunk 5]
```diff
@@ -398,8 +398,8 @@ def persist(*args, **kwargs):
 
     For example a lazy dask.array built up from many lazy calls will now be a
     dask.array of the same shape, dtype, chunks, etc., but now with all of
-    those previously lazy tasks either computed in memory as many small NumPy
-    arrays (in the single-machine case) or asynchronously running in the
+    those previously lazy tasks either computed in memory as many small :function:`numpy.array`
```
Comment (TomAugspurger) [inline dask/base.py]: Same comment about ndarray.

File: docs/README.rst [hunk 6]
```diff
@@ -6,5 +6,6 @@ these commands suffice::
   cd dask/docs
   conda create -n daskdocs --file requirements-docs.txt
   source activate daskdocs
+  Update the conf.py file with the code path (e.g. sys.path.insert(0, os.path.abspath('../../')) )
```
Comment (TomAugspurger) [inline docs/README.rst]: Could you try doing an editable install from the root of the `dask` repo?

    python -m pip install -e .

That should avoid having to modify the conf file.

File: dask/array/core.py (hunk 1 above)
Comment (sephib) [reply dask/array/core.py]: `:func:`  works, any other options doesn't  generate a link

**Comment by sephib** [comment]
Following @TomAugspurger  comments I've updated the files

**Comment by mrocklin** [comment]
cc @TomAugspurger any further comments here?

**Comment by TomAugspurger** [comment]
Looks good, thanks.

**Pull request merged**